airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/jsonl/JsonLSerializedBuffer.kt

```diff
@@ -41,11 +43,28 @@ class JsonLSerializedBuffer(
         printWriter = PrintWriter(outputStream, true, StandardCharsets.UTF_8)
     }
 
-    @Deprecated("Deprecated in Java")
-    override fun writeRecord(record: AirbyteRecordMessage) {
-        val json = MAPPER.createObjectNode()
+    private fun addV2Fields(json: ObjectNode, record: AirbyteRecordMessage, generationId: Long) {
+        json.put(JavaBaseConstants.COLUMN_NAME_AB_RAW_ID, UUID.randomUUID().toString())
+        json.put(JavaBaseConstants.COLUMN_NAME_AB_EXTRACTED_AT, record.emittedAt)
+        json.put(JavaBaseConstants.COLUMN_NAME_AB_LOADED_AT, System.currentTimeMillis())
+        json.put(JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID, generationId)
+        json.replace(JavaBaseConstants.COLUMN_NAME_AB_META, MAPPER.valueToTree(record.meta))
+    }
+
+    private fun addV1Fields(json: ObjectNode, record: AirbyteRecordMessage) {
         json.put(JavaBaseConstants.COLUMN_NAME_AB_ID, UUID.randomUUID().toString())
         json.put(JavaBaseConstants.COLUMN_NAME_EMITTED_AT, record.emittedAt)
+    }
+
+    @Deprecated("Deprecated in Java")
```

Comment:
.... I know you didn't write this, but I assume this should just be deprecated everywhere? I'm assuming this method was deprecated before we introduced kotlin >.>